src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -34640,6 +34640,519 @@ .maxstack  2
+                using System.Collections;
+                using System.Collections.Generic;
+
+                class MyCollection(List<int> list) : IEnumerable<int>
+                {
+                    public IEnumerator<int> GetEnumerator() => list.GetEnumerator();
+
+                    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+                }
+
+                class C
+                {
+                    static void Main()
+                    {
+                        M(new([1, 2, 3])).Report();
+                    }
+
+                    static List<int> M(MyCollection c) => [..c];
+                }
+                """;
+
+            var verifier = CompileAndVerify([source, s_collectionExtensionsWithSpan], expectedOutput: IncludeExpectedOutput("[1, 2, 3],"), targetFramework: TargetFramework.Net80, verify: Verification.Skipped);
+            verifier.VerifyDiagnostics();
+
+            verifier.VerifyIL("C.M", """
+                {
+                  // Code size       13 (0xd)
+                  .maxstack  3
+                  IL_0000:  newobj     "System.Collections.Generic.List<int>..ctor()"
+                  IL_0005:  dup
+                  IL_0006:  ldarg.0
+                  IL_0007:  callvirt   "void System.Collections.Generic.List<int>.AddRange(System.Collections.Generic.IEnumerable<int>)"
+                  IL_000c:  ret
+                }
+                """);
+        }
+
+        [Fact]
+        public void List_SingleSpread_CustomCollection_ICollectionAndNoStructEnumerator()
+        {
+            var source = """
+                using System.Collections;
+                using System.Collections.Generic;
+
+                class MyCollection(List<int> list) : ICollection<int>
+                {
+                    public int Count => list.Count;
+
+                    public bool IsReadOnly => false;
+
+                    public void Add(int item) => list.Add(item);
+
+                    public void Clear() => list.Clear();
+
+                    public bool Contains(int item) => list.Contains(item);
+
+                    public void CopyTo(int[] array, int arrayIndex) => list.CopyTo(array, arrayIndex);
+
+                    public bool Remove(int item) => list.Remove(item);
+
+                    public IEnumerator<int> GetEnumerator() => list.GetEnumerator();
+
+                    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+                }
+
+                class C
+                {
+                    static void Main()
+                    {
+                        M(new([1, 2, 3])).Report();
+                    }
+
+                    static List<int> M(MyCollection c) => [..c];
+                }
+                """;
+
+            var verifier = CompileAndVerify([source, s_collectionExtensionsWithSpan], expectedOutput: IncludeExpectedOutput("[1, 2, 3],"), targetFramework: TargetFramework.Net80, verify: Verification.Skipped);
```

Comment:
>var verifier = CompileAndVerify([source, s_collectionExtensionsWithSpan], expectedOutput: IncludeExpectedOutput("[1, 2, 3],"), targetFramework: TargetFramework.Net80, verify: Verification.Skipped); [](http://example.com/codeflow?start=12&length=197)

Consider also compiling and verifying IL for the default target framework. That should avoid the `CollectionsMarshal` optimization and use `AddRange()` instead. And it should be possible to test `expectedOutput` regardless of platform for that case. Please consider adding to this test rather than creating a separate test with duplicated source.
```
verifier = CompileAndVerify([source, s_collectionExtensions], expectedOutput: "[1, 2, 3],", verify: Verification.Skipped);
verifier.VerifyIL("C.M", ...);
```

Reply:
> And it should be possible to test `expectedOutput` regardless of platform for that case

Unfortunatelly, it is not. When tests are run on .NET Framework, it cannot load and execute code, generated for .NET 8